Source/ARTDevicePushDetails.m

```diff
@@ -20,8 +21,35 @@ - (id)copyWithZone:(NSZone *)zone {
     return push;
 }
 
+- (NSString *)stateString {
+    switch (_state) {
+        case ARTPushStateActive:
+            return @"ARTPushStateActive";
+        case ARTPushStateFailing:
+            return @"ARTPushStateFailing";
+        case ARTPushStateFailed:
+            return @"ARTPushStateFailed";
+        default:
+            return @"ARTPushStateUnknown";
+    }
+}
+
++ (ARTPushState)stateFromString:(NSString *)string {
```

Comment:
Probably just naming, but I expect to use here values from `stateString` like `ARTPushStateActive`, `ARTPushStateFailing` etc.
Maybe additionally checking these values in this method could be a better idea than renaming this method.

Reply:
Hmm, I'm not sure what you are saying here, this method is for obtaining `ARTPushState` values from strings that came from `JSON` data. Am I missing something?

Reply:
Ok, so you should rename it to `stateFromJsonString`.
I meant you have this method:
```swift
- (NSString *)stateString {
    switch (_state) {
        case ARTPushStateActive:
            return @"ARTPushStateActive";
        case ARTPushStateFailing:
            return @"ARTPushStateFailing";
        case ARTPushStateFailed:
            return @"ARTPushStateFailed";
        default:
            return @"ARTPushStateUnknown";
    }
}
```
which also produce strings from `ARTPushState` and the method name `stateFromString` is a bit confusing because I don't know "from which string" it creates the `ARTPushState`

Reply:
good point https://github.com/ably/ably-cocoa/pull/1186/commits/b03a25126508a433e6025a8f656538dd865ac563